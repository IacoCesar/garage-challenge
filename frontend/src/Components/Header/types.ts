export interface IHeader {
  testID?: string
  isHomeScreen?: boolean
  title?: string
  callHandleBack?: (navigation: any) => void
  callHandleChangeVisualMode?: (themeSelector: string) => void
}

export interface IProps extends IHeader {
  themeSelector: string
}
