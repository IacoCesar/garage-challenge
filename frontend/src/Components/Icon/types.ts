export interface IProps {
  testID?: string
  size: number
  name: string
  color: string
  type?: string
}
