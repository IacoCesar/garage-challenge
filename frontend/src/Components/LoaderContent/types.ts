export interface IProps {
  testID?: string
  size: number
  color: string
}
