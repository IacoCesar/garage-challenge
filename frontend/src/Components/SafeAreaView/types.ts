export interface IProps {
  testID?: string
  children: JSX.Element
}

export interface IThemeProps {
  themeSelector: string
}
