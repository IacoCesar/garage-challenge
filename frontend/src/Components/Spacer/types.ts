export interface IProps {
  testID?: string
  min: number
  max: number
}
