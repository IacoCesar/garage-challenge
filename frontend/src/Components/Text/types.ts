export interface ITextProps {
  color?: string
  fontSize?: number
  isBold?: boolean
}

export interface IProps extends ITextProps {
  label: string
  testID?: string
}
