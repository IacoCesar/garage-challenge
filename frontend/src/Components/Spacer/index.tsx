import { assignTestId } from '@/Helpers/TestId'
import React from 'react'
import S from './styles'
import { IProps } from './types'

const SpacerView: React.FC<IProps> = ({ testID = 'Spacer', min, max }) => (
  <S.Spacer {...assignTestId('View', `${testID}`)} min={min} max={max} />
)

export default SpacerView
