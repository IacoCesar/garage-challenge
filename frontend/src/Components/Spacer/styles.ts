import styled from 'styled-components/native'
import { IProps } from './types'

const Spacer = styled.View<IProps>`
  flex-grow: 1;
  ${({ min }) => min && `min-height: ${min}px`};
  ${({ max }) => max && `max-height: ${max}px`};
`

export default { Spacer }
