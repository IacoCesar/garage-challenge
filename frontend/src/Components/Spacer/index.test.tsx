import React from 'react'
import { IProps } from './types'
import { mockTestId } from '@/Helpers/TestId'
import Spacer from '.'
import { render } from '@/Helpers/TestHelpers'

describe('Spacer', () => {
  let props: IProps = {
    testID: 'Spacer',
    min: 20,
    max: 20
  }

  const loaderElement = mockTestId('View', `${props.testID!}`).testID

  it(`should render ${loaderElement}`, async () => {
    const wrapper = render(<Spacer {...props} />)

    const el_container = wrapper.getByTestId(loaderElement)
    expect(el_container).toBeTruthy()
  })

  it('should render snapshot', async () => {
    const wrapper = render(<Spacer {...props} />)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
