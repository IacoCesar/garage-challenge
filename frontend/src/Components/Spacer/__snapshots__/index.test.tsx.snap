// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer should render snapshot 1`] = `
<View
  accessible={false}
  max={20}
  min={20}
  style={
    Array [
      Object {
        "flexGrow": 1,
        "maxHeight": 20,
        "minHeight": 20,
      },
    ]
  }
  testID="View_Spacer"
/>
`;
